bentoml/cli/bento_service.py

```diff
@@ -0,0 +1,536 @@
+        help="Install shell command completion",
+        short_help="Install shell command completion",
+    )
+    @click.option(
+        '--append/--overwrite',
+        help="Append the completion code to the file",
+        default=None,
+    )
+    @click.argument('shell', required=False, type=shell_types)
+    @click.argument('path', required=False)
+    def install_completion(append, shell, path):
+        if click_completion:
+            # click_completion package is imported
+            shell, path = click_completion.core.install(
+                shell=shell, path=path, append=append
+            )
+            click.echo('%s completion installed in %s' % (shell, path))
+        else:
+            click.echo(
+                "'click_completion' is required for BentoML auto-completion, "
+                "install it with `pip install click_completion`"
+            )
+
+    @bentoml_cli.command(
+        help='Containerizes given Bento into a ready-to-use Docker image.',
+        short_help="Containerizes given Bento into a ready-to-use Docker image",
+    )
+    @click.argument("bento", type=click.STRING)
+    @click.option('-p', '--push', is_flag=True)
+    @click.option(
+        '-t',
+        '--tag',
+        help="Optional image tag. If not specified, Bento will generate one from "
+        "the name of the Bento.",
+        required=False,
+    )
+    @click.option(
+        '-u', '--username', type=click.STRING, required=False,
+    )
+    @click.option(
+        '-p', '--password', type=click.STRING, required=False,
+    )
+    def containerize(bento, push, tag, username, password):
+        """Containerize specified BentoService.
+
+        BENTO is the target BentoService to be containerized, referenced by its name
+        and version in format of name:version. For example: "iris_classifier:v1.2.0"
+
+        `bentoml containerize` command also supports the use of the `latest` tag
+        which will automatically use the last built version of your Bento.
+
+        You can provide a tag for the image built by Bento using the
+        `--docker-image-tag` flag. Additionally, you can provide a `--push` flag,
+        which will push the built image to the Docker repository specified by the
+        image tag.
+
+        You can also prefixing the tag with a hostname for the repository you wish
+        to push to.
+        e.g. `bentoml containerize IrisClassifier:latest --push --tag username/iris`
+        would build a Docker image called `username/iris:latest` and push that to
+        Docker Hub.
+
+        By default, the `containerize` command will use the credentials provided by
+        Docker. You may provide your own through `--username` and `--password`.
+        """
+        bento_service_bundle_path = resolve_bundle_path(
+            bento, pip_installed_bundle_path
+        )
+
+        _echo(f"Found Bento: {bento_service_bundle_path}")
+
+        name, version = get_name_version_from_tag(bento)
```

Comment:
This will break when the `bento` parameter is a file path. You can get the name and version with `load_bento_service_metadata(bento_service_bundle_path)`

Reply:
This will allow us to use the containerize by giving it the saved_path:

`bentoml containerize $SAVED_PATH -t test_image`

Reply:
Currently it breaks with the follow error:
```
 $  bentoml containerize /Users/chaoyu/bentoml/repository/IrisClassifier/20200701222443_EC5EAB -t test_image
Found Bento: /Users/chaoyu/bentoml/repository/IrisClassifier/20200701222443_EC5EAB
Traceback (most recent call last):
  File "/Users/chaoyu/opt/miniconda3/envs/btml-dev/bin/bentoml", line 33, in <module>
    sys.exit(load_entry_point('BentoML', 'console_scripts', 'bentoml')())
  File "/Users/chaoyu/opt/miniconda3/envs/btml-dev/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/chaoyu/opt/miniconda3/envs/btml-dev/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/chaoyu/opt/miniconda3/envs/btml-dev/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/chaoyu/opt/miniconda3/envs/btml-dev/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/chaoyu/opt/miniconda3/envs/btml-dev/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/chaoyu/workspace/BentoML/bentoml/cli/click_utils.py", line 140, in wrapper
    return func(*args, **kwargs)
  File "/Users/chaoyu/workspace/BentoML/bentoml/cli/click_utils.py", line 117, in wrapper
    return_value = func(*args, **kwargs)
  File "/Users/chaoyu/workspace/BentoML/bentoml/cli/click_utils.py", line 101, in wrapper
    return func(*args, **kwargs)
  File "/Users/chaoyu/workspace/BentoML/bentoml/cli/bento_service.py", line 484, in containerize
    name, version = make_bento_name_docker_compatible(name, version)
  File "/Users/chaoyu/workspace/BentoML/bentoml/cli/bento_service.py", line 110, in make_bento_name_docker_compatible
    tag = tag.lstrip(".-")[:128]
AttributeError: 'NoneType' object has no attribute 'lstrip'
```